name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/Extractor.java

```diff
@@ -441,12 +441,12 @@ public String getSource()
         public Status apply(ImportAction action, Context context)
         {
             Account account = getAccountPrimary();
-            if (account == null)
-                account = context.getAccount();
+            if (account == null && getAmount() != null)
+                account = context.getAccount(getAmount().getCurrencyCode());
 
             Account accountSecondary = getAccountSecondary();
-            if (accountSecondary == null)
-                accountSecondary = context.getSecondaryAccount();
+            if (accountSecondary == null && getAmount() != null)
+                accountSecondary = context.getSecondaryAccount(getAmount().getCurrencyCode());
```

Comment:
I believe the account has to be looked up by the target account currency. If you use #getAmount(), this will be the currency of the source account. But the user cannot configure target accounts based on different source account currencies.